src/main/java/org/tarantool/RoundRobinSocketProviderImpl.java

```diff
@@ -117,8 +120,26 @@ protected InetSocketAddress getLastObtainedAddress() {
     }
 
     @Override
-    protected InetSocketAddress getAddress(int retryNumber, Throwable lastError) throws IOException {
-        return getNextSocketAddress();
+    protected SocketChannel makeAttempt(int retryNumber, Throwable lastError) throws IOException {
```

Comment:
I would clarify in a comment that we doing config.retriesCount connect attempts to one isntance within one `makeAttempt()` call and that the following calls will try the next addresses.

It is obvious from the code, yep, but I was a bit confused when look the first time. Maybe it is because retryNumber and retriesLimit are something about different things.